### ISSUE TITLE
_EPS and _NUDGE are scaled to different things, so their ratio is unbounded — CACTUS-scale meshes land in the regime where 24% of walkers escape

### ISSUE THREAD

**Issue by rutgerfick** [dmrai-lab/dmipy-sim#53]
`Mesh` carries two length constants that guard the collision test, and they are scaled to different things:

```python
_scale = float(np.min(self.L))    # the BOX
self._EPS   = 1e-7 * _scale       # smallest resolvable length in float32 — domain-scaled, correctly
self._NUDGE = 1e-4 * step_l       # how far off the wall a reflected walker is parked — step-scaled
```

`_EPS` discards a candidate collision closer than the float32 noise floor, which is how the walker avoids
re-detecting the triangle it just left. `_NUDGE` is the clearance that guard has to see past. Nothing ties
them together, so their ratio is a free variable of the geometry:

```
_EPS/_NUDGE  =  6e-3 * box / feature_radius
```

Once the ratio reaches 1, the guard is wider than the clearance: a genuine wall lying just ahead of a
freshly-reflected walker is discarded as self-intersection, and the walker passes through it.

## It is reachable with ordinary parameters

| substrate | box | feature radius | `_EPS/_NUDGE` |
|---|---|---|---|
| Winther axon06 intra | 10 µm | 0.60 µm | 0.10 |
| test cylinder (subdiv 2) | 6 µm | 0.375 µm | 0.096 |
| CACTUS, `fr = min_rad` | 30 µm | 0.27 µm | 0.67 |
| **CACTUS, `fr = min_rad/2`** | **30 µm** | **0.135 µm** | **1.33** |

CACTUS substrates are generated with `min_rad 0.27` in a `lenght_side 30` box (`prod/cross45.txt`), so a
mesh resolved to half its thinnest feature — an ordinary choice — lands past the threshold. A fine mesh in
a large box is exactly the regime the bank is moving toward.

## Measured sensitivity

Closed cylinder, 20,000 walkers, `reject_escape` off, escape counted beyond 1 nm of depth:

| `_EPS/_NUDGE` | escaped |
|---|---|
| 0.0016 (`_EPS` × 0.01) | 2 / 20,000 |
| 0.160 (shipped) | 6 / 20,000 |
| 16.0 (`_EPS` × 100) | **4,811 / 20,000 — 24%** |

## Fix

Floor the nudge against the noise floor it has to clear, so the ratio is bounded whatever the mesh does:

```python
self._NUDGE = jnp.float32(max(1e-4 * step_l, 64.0 * float(self._EPS)))
```

That pins the ratio at 1/64 = 0.0156 at every resolution tested (0, 1, 2, 3 subdivisions), and the nudge
stays at 1.2e-4 to 5.1e-4 of a step length — orders of magnitude below Monte-Carlo noise on any observable.

**This is a hazard guard, not a fix for #51.** At the shipped ratio it moves the escape count from 6/20,000
to 4/20,000, which is ~1σ and not evidence of anything. What it does is make the catastrophic regime
unreachable, which today it is not.

**Comment by rutgerfick** [comment]
## Measured at the CACTUS ratio

Same harness, 4,000 walkers, closed cylinder, `reject_escape` off, escape counted beyond 1 nm of depth:

| `_EPS/_NUDGE` | escaped / 4,000 |
|---|---|
| 0.160 (shipped) | 0 (0.000%) |
| **1.333 (what CACTUS parameters give)** | **38 (0.950%)** |

So the regime is not merely theoretical and not merely a tail: at the ratio a CACTUS-resolved mesh lands on, **~1% of intra walkers leave the axon**. For comparison the Monte-Carlo floor of the packs we ship is 9.33e-3 median, so a 1% walker loss is at or above the noise each pack certifies against — it would bias the intra signal and `f_intra` rather than disappear into scatter.

This puts the fix on the critical path for simulating CACTUS substrates, alongside the meshes themselves.

Interpolating the three points (0.16 → 0%, 1.33 → 0.95%, 16 → 24%), the ratio wants to stay well under 1; the proposed floor of `64 * _EPS` puts it at 0.0156 for every mesh.
